src/core/SegmentedControl/index.tsx

```diff
@@ -0,0 +1,71 @@
+import {
+  ButtonGroupProps,
+  IconButtonProps as RawIconButtonProps,
+} from "@material-ui/core";
+import ToggleButton from "@material-ui/lab/ToggleButton";
+import React from "react";
+import Icon, { IconNameToSizes } from "../Icon";
+import Tooltip from "../Tooltip";
+import { StyledSegmentedControl } from "./style";
+// one prop is array of objects: with icon name and tooltip text. They need to make
+// first item in array first button, etc
+interface SingleButtonDefinition {
+  iconName: string;
+  tooltipText: string;
+}
+
+interface ExtraProps {
+  buttonDefinition: SingleButtonDefinition[];
+}
+
+export type SegmentedControlProps = ExtraProps &
+  ButtonGroupProps &
+  RawIconButtonProps;
+
+// function(event: React.MouseEvent<HTMLElement>, value: any) => void
+
+const SegmentedControl = (props: SegmentedControlProps) => {
+  const { buttonDefinition } = props;
+  const leftmost = buttonDefinition[0]?.tooltipText;
+  const [active, setActive] = React.useState<string | null>(leftmost);
+
+  const handleActive = (
+    event: React.MouseEvent<HTMLElement>,
+    newActive: string | null
+  ) => {
+    if (newActive !== null) {
+      setActive(newActive);
+    }
+  };
+
+  return (
+    <StyledSegmentedControl
+      {...props}
```

Comment:
I think `{...props}` normally goes at the end in case the consumer wants to overwrite some of the sds defined props with something custom. Is there a reason this is first, here?